src/aliceVision/keyframe/KeyframeSelector.cpp

```
@@ -464,80 +464,89 @@ bool KeyframeSelector::writeSelection(const std::vector<std::string>& brands,
+                if (feed.isVideo()) {
+                    options.fromColorSpace(image::EImageColorSpace::SRGB);
                     options.toColorSpace(image::EImageColorSpace::AUTO);
-            }
+                } else {  // Otherwise, the frames have been read without any conversion, they should be written as such
+                    if (colorspace == "sRGB")
+                            options.fromColorSpace(image::EImageColorSpace::SRGB);
+
+                    if (outputExtension == "exr")
+                        options.toColorSpace(image::EImageColorSpace::NO_CONVERSION);
+                    else
+                        options.toColorSpace(image::EImageColorSpace::AUTO);
+                }
 
-            if (storageDataType != image::EStorageDataType::Undefined && outputExtension == "exr"){
-                options.storageDataType(storageDataType);
-            }
+                if (storageDataType != image::EStorageDataType::Undefined && outputExtension == "exr") {
+                    options.storageDataType(storageDataType);
+                }
 
-            image::writeImage(filepath, image, options, metadata);
-            ALICEVISION_LOG_DEBUG("Wrote selected keyframe " << pos);
+                image::writeImage(filepath, image, options, metadata);
+                ALICEVISION_LOG_DEBUG("Wrote selected keyframe " << pos);
 
-            _keyframesPaths[id].push_back(filepath);
+                _keyframesPaths[id].push_back(filepath);
+            }
         }
 
+        // If the current media is a video and there is no output keyframe, the corresponding SfMData file will not be written
+        if (feed.isVideo() && outputExtension == "none") {
+            ALICEVISION_LOG_WARNING("The keyframes selected from the input video have not been written on disk. " <<
+                                    "The keyframes' SfMData file cannot be written.");
+            continue;
```

We probably want to throw an error in that case.